src/data_api/reporting.se

```diff
@@ -27,186 +27,161 @@ extern branches: [addCurrency:[int256,int256,int256,int256]:int256, addMarketToB
-    self.Reporting[1010101].repIDtoIndex[msg.sender] = 1
-    self.Reporting[1010101].activeRep += redistributionRep
-    self.Reporting[1010101].totalRep += redistributionRep
-
-def getReportedOnNonFinalRoundTwoEvent(branch, reporter):
-    return(self.Reporting[branch].reportedOnNonFinalRoundTwoEvent[reporter])
-
-def setReportedOnNonFinalRoundTwoEvent(branch, event, reporter):
+    self.Reporting[1010101].activeRep[0x0000000000000000000000000000000000000000] = 0
+    self.Reporting[1010101].dormantRep[0x0000000000000000000000000000000000000000] = redistributionRep
+    self.Reporting[1010101].totalActiveRep = 0
+    self.Reporting[1010101].totalDormantRep = redistributionRep
+    # TODO: figure out why address 0 counts as a reporter, feels like this should be `0`
+    self.Reporting[1010101].numberReporters = 1
+    self.Reporting[1010101].reporterLocks[0x0000000000000000000000000000000000000000].eventCount = 0
+
+def getNumberOfEventsLockingReporter(branch, reporterAddress):
+    return(self.Reporting[branch].reporterLocks[reporterAddress].eventCount)
+
+def lockReporter(branch, reporterAddress, event):
     self.controller.checkWhitelist(msg.sender)
-    self.Reporting[branch].reportedOnNonFinalRoundTwoEvent[reporter] = event
+    self.Reporting[branch].reporterLocks[reporterAddress].events[event] = 1
+    self.Reporting[branch].reporterLocks[reporterAddress].eventCount += 1
     return(1)
 
-# @return fxp
-def getActiveRep(branch):
-    return(self.Reporting[branch].activeRep)
+def unlockReporter(branch, reporterAddress, event):
+    self.controller.checkWhitelist(msg.sender)
+    if (self.Reporting[branch].reporterLocks[reporterAddress].events[event] == 0):
+        throw()
+    if (self.Reporting[branch].reporterLocks[reporterAddress].eventCount <= 0):
+        throw() # if we are in this state things are _really_ broken
+    self.Reporting[branch].reporterLocks[reporterAddress].events[event] = 0
+    self.Reporting[branch].reporterLocks[reporterAddress].eventCount -= 1
+    return(1)
 
-def adjustActiveRep(branch, fxpAmount):
+def activateRep(branch, attorep):
+    return(self.activateRepFor(branch, attorep, msg.sender))
+
+def activateRepFor(branch, attorepToActivate, reporterAddress):
     self.controller.checkWhitelist(msg.sender)
-    self.Reporting[branch].activeRep += fxpAmount
+    # CONSIDER: should people who are locked be prevented from activating rep?
+    startingDormantRep = self.Reputation[branch].dormantRep[reporterAddress]
+    startingActiveRep = self.Reputation[branch].activeRep[reporterAddress]
+    self.Reputation[branch].dormantRep[reporterAddress] = safeSub(startingDormantRep, attorepToActivate)
+    self.Reputation[branch].activeRep[reporterAddress] = safeAdd(startingActiveRep, attorepToActivate)
     return(1)
 
-# @return fxp
-def getRepByIndex(branch, repIndex):
-    return(self.Reporting[branch].reputation[repIndex].repValue)
+def deactivateRep(branch, attorep):
```

Comment:
ditto for this one re removing, any account that holds rep will never be calling deactivateRep and so the msg.sender will never be right